tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Core.cs

```diff
@@ -82,7 +82,7 @@ internal void ReportAndBlock(IResult? result)
         }
     }
 
-    private Dictionary<string, object> GetBasicRequestArgsForWaf()
+    private Dictionary<string, object?> GetBasicRequestArgsForWaf()
```

Comment:
In a lot of places below, we try hard to make sure the values _don't_ return `null`. I would suggest that we should either harden this method to make sure it's always non-null, or remove those fallbacks (`?? string.Empty`) that were added to achieve that?

Also, I would  recommend making sure there's at least a unit test or an integration test that ends up putting null in here, just to make sure it definitely _is_ ok (given I seem to remember we guarded these originally because it causes exceptions, but I could be wrong)